### PR TITLE
feat: add logo url for wallet response

### PR DESCRIPTION
**Pull request by pietro-tota**
Depends on https://github.com/pagopa/pagopa-infra/pull/1800

#### List of Changes
Add configurable logos CDN URI to be returned to the caller when retrieving wallet.
This configurable map will perform checks against configuration checking that all expected wallet types are correctly configured and raising an exception in case of missing required type-logo mapping, prevent module startup

#### Motivation and Context
This logic is required in order to return wallet proper logo based on wallet type (card brand logos for card wallet, apm logo for apm wallets and so on)

#### How Has This Been Tested?
Junit tests

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/kotlin/it/pagopa/wallet/config/LogoConfig.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    ): Map<String, URI> {
+        val transformedLogoMapping = logoMapping.mapValues { URI.create(it.value) }
+        val expectedKeys =
+            (WalletCardDetailsDto.BrandEnum.values().map { it.toString() } +
+                    WalletDetailsType.values()
+                        .filter { it != WalletDetailsType.CARDS }
+                        .map { it.toString() } +
+                    LogoUtils.UNKNOWN_LOGO_KEY)
+                .toSet()
+        val missingKeys = expectedKeys.filter { !transformedLogoMapping.containsKey(it) }
```
Comment (giovanniberti) [inline src/main/kotlin/it/pagopa/wallet/config/LogoConfig.kt:26]:
```suggestion
        val missingKeys = expectedKeys - transformedLogoMapping.keys
```

File: src/main/kotlin/it/pagopa/wallet/util/LogoUtils.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import it.pagopa.wallet.domain.wallets.Wallet
+import it.pagopa.wallet.domain.wallets.details.CardDetails
+import it.pagopa.wallet.domain.wallets.details.PayPalDetails
+import java.net.URI
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+@Component
+class LogoUtils(@Autowired val logoMapping: Map<String, URI>) {
```
Comment (giovanniberti) [inline src/main/kotlin/it/pagopa/wallet/util/LogoUtils.kt:11]: Don't love the name but can't think of a much better one 🤔
Maybe `LogoResources`? Idk, don't love that either, still posting a comment to spark a discussion

File: src/test/kotlin/it/pagopa/wallet/WalletTestUtils.kt [hunk 3]
```diff
@@ -219,7 +220,10 @@ object WalletTestUtils {
             )
     }
 
-    fun walletDocumentStatusValidatedCARD(): Wallet {
+    fun walletDocumentStatusValidatedCARD() =
```
Comment (giovanniberti) [inline src/test/kotlin/it/pagopa/wallet/WalletTestUtils.kt:223]:
```suggestion
    fun walletDocumentStatusValidatedCard() =
```

**Review by giovanniberti — DISMISSED** [review]
Just minor issues :)

File: src/main/kotlin/it/pagopa/wallet/config/LogoConfig.kt [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+
+@Configuration
+class LogoConfig {
+
+    @Bean
+    fun walletLogoMapping(
+        @Value("#{\${wallet.logo_mapping}}") logoMapping: Map<String, String>
```
[the remark this answers is not in the document]
Comment (pietro-tota) [reply src/main/kotlin/it/pagopa/wallet/config/LogoConfig.kt:16]: this is the way to make spring parse this configuration value as a map (using spel expression). removing this makes a runtime error where spring complains about the fact it cannot cast string to Map

File: src/main/kotlin/it/pagopa/wallet/util/LogoUtils.kt (hunk 2 above)
Comment (pietro-tota) [reply src/main/kotlin/it/pagopa/wallet/util/LogoUtils.kt:11]: What about `WalletsLogoUtils`?

File: src/test/kotlin/it/pagopa/wallet/WalletTestUtils.kt (hunk 3 above)
Comment (pietro-tota) [reply src/test/kotlin/it/pagopa/wallet/WalletTestUtils.kt]: fixed with https://github.com/pagopa/pagopa-wallet-service/pull/100/commits/4c42a49ef00c94f19e7f532816179937825b2344

File: api-spec/wallet-api.yaml [hunk 5]
```diff
@@ -939,6 +939,10 @@ components:
             mapping:
               CARDS: '#/components/schemas/WalletCardDetails'
               PAYPAL: '#/components/schemas/WalletPaypalDetails'
+        logoUrl:
+          description: url for the logo associated to the wallet
+          type: string
+          format: uri
```
[the remark this answers is not in the document]
Comment (pietro-tota) [reply api-spec/wallet-api.yaml:945]: idk, maybe asset is too generic as name, logoUrl does not leave any space to misunderstanding. What do you think? I would leave it as logoUrl imho

**Review by giovanniberti — APPROVED** [review]

File: src/main/kotlin/it/pagopa/wallet/util/LogoUtils.kt (hunk 2 above)
Comment (pietro-tota) [reply src/main/kotlin/it/pagopa/wallet/util/LogoUtils.kt]: Refactored to WalletUtils with https://github.com/pagopa/pagopa-wallet-service/pull/100/commits/98478a93356237d33d775a85c9b25ec4741cc5f4

**Review by Fernando-Granato — DISMISSED** [review]

**Review by infantesimone — DISMISSED** [review]

File: api-spec/wallet-api.yaml [hunk 6]
```diff
@@ -994,7 +998,8 @@ components:
         - creationDate
         - updateDate
         - type
-        - applications
+        - services
```
Comment (pietro-tota) [inline api-spec/wallet-api.yaml:1001]:
```suggestion
        - applications
```

**Review by infantesimone — APPROVED** [review]

**Pull request merged**